src/message/messageSelectors.js

```diff
@@ -2,10 +2,22 @@
 import { createSelector } from 'reselect';
 
 import type { Narrow, RenderedSectionDescriptor, Selector } from '../types';
-import { getFlags, getMute, getSubscriptions } from '../directSelectors';
+import {
+  getAllNarrows,
+  getFlags,
+  getMessages,
+  getMute,
+  getSubscriptions,
+} from '../directSelectors';
 import { getShownMessagesForNarrow } from '../chat/narrowsSelectors';
 import renderMessages from './renderMessages';
 import { findAnchor } from '../utils/message';
+import { ALL_PRIVATE_NARROW_STR } from '../utils/narrow';
+import { NULL_ARRAY } from '../nullObjects';
+
+export const getPrivateMessages = createSelector(getAllNarrows, getMessages, (narrows, messages) =>
+  (narrows[ALL_PRIVATE_NARROW_STR] || NULL_ARRAY).map(id => messages[id]),
```

Comment:
Are there any significant advantages to using `NULL_ARRAY` / `Object.freeze([])` here instead of just an array literal? It seems like it's an unnecessary layer here or I might be missing something.

Reply:
Not sure how significant, but yeah. Here are few:
* Saves some memory (maybe too little to matter)
* Can not be mutated
* Most importantly `a === b` where both are `NULL_ARRAY` but `a !== b` when initialized with `[]`